### DISCUSSION TITLE
How to send a JSON POST payload with JS client, and read with hook?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#3367] [Q&A] [answered]
I'd like to use the Pocketbase JS SDK client to send a `POST` with a `JSON` payload in the body, then I'd like to read it in a hook. I looked at https://github.com/pocketbase/js-sdk#usage and https://pocketbase.io/docs/js-routing/#reading-request-body, but I don't see examples of how to send and receive in this style.

Can you please provide an example?

```ts
client.send(`POST`, { body: someJson })
```

```ts
routerAdd(`POST`, c=> {
   const rawBody =    // read the raw JSON body?

   const data = JSON.parse(rawBody)
})
```

**Comment by ganigeorgiev** [comment, marked as answer]
If you need the raw request body as plain string, you can use [`readerToString(c.request().body)`](https://pocketbase.io/jsvm/functions/readerToString.html) (_this method is available with [PocketBase v0.18.3+](https://github.com/pocketbase/pocketbase/releases/tag/v0.18.3); other helper methods for the writer/reader Go interface will be available with the next v0.19/v0.20 releases_).

Alternatively, you can also check https://pocketbase.io/docs/js-routing/#reading-request-body.

**Reply by ganigeorgiev** [reply]
_Side-note - for the JS SDK client call the format should be `pb.send(path, { method: "POST", body: formData })`, see https://pocketbase.io/docs/js-routing/#sending-request-to-custom-routes-using-the-sdks._

**Reply by cheskoxd** [reply]
Thanks a lot, been searching for this on the https://pocketbase.io/jsvm and couldnt find it
